File: FancyDelivery/Source/FancyDeliveryApp/Scripts/app/product.js
function ProductViewModel(productData) {
    var self = this;
    self.Name = ko.observable(productData.Name);
    self.Id = ko.observable(productData.Id);
    self.Price = ko.observable(productData.Price);
    self.ImageName = ko.observable(productData.ImageName);
    self.add_to_cart = function(product) {
        mainViewModel.AddProductToCart(product);
    };
}

File: FancyDelivery/Source/Infrastructure/Models/Product.cs
using System;

namespace Infrastructure
{
    public class Product
    {
        public virtual int Id { get; set; }
        public virtual string Name { get; set; }
        public virtual string Description { get; set; }
        public virtual Category Category { get; set; }
        public virtual string ImageName { get; set; }
        public virtual Decimal Price { get; set; }
    }
}